### ISSUE TITLE
Update CDM (in place) webdav implementation so it gracefully handles network disconnects

### ISSUE THREAD

**Issue by Kjames5269** [codice/ddf#4268]
### Description
Upon a network disconnect the current webdav implementation will stop all processing when an error occurs and throws an error to the console. The problem occurs that the state, at least when the metacards are being created, only updates once the entire operation finishes. This can lead to inconsistent states where multiple metacards exist per webdav entry. This is also a problem with delete and potentially a problem for modifications. 

### Steps to Reproduce (if issue is bug)
1. Start up a webdav server, (follow the directions [here](https://codice.atlassian.net/browse/DDF-3372) to set it up over Tomcat)
2. set a breakpoint in the code @ [DavAlterationObserver#177](https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DavAlterationObserver.java#L177)
2. Set up CDM to point to the webdav server
3. Ingest one or two metacards and then shutdown the tomcat server. (Press play one or two times)
4. wait until the error is thrown
5. Start the tomcat server again
6. Query for the metacard, and note there are two of them.

#### Expected behavior:
The state gets updated after a process successfully completes.
There is only one metacard after a disconnect and reconnect.

#### Actual behavior:
The state gets updated at the end of the batch.
There are multiple metacards after a disconnect and reconnect

#### Reproduces how often:
100%

### Version
This is a problem on both the [2.13.x], and [master] branches, along with any branch cut from those.

### Additional Information
Any additional information, configuration or data that might be necessary
to reproduce the issue.

**Comment by Kjames5269** [comment]
apologizes on the commit message, these changes will depend upon #4223

**Comment by Kjames5269** [comment]
Stopping work on this issue.

**Comment by shaundmorris** [comment]
Closing this issue since work has stopped on it over a year ago.
